File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_ = sum_ = sum(nums[:k])
        for i, n in enumerate(nums[k:]):
            sum_ += n - nums[i]
            if sum_ > max_:
                max_ = sum_
        return max_/k

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        long double sum_;
        int N = k;
        if (nums.size() < k) N = nums.size();
        int i = 0;
        for (; i < N; ++i){
            sum_ += nums[i];
        }
        long double maxSum = sum_;
        int l = 0;
        for (; i < nums.size(); ++i, ++l){
            sum_ += nums[i] - nums[l];
            if (sum_ > maxSum) maxSum = sum_;
        }
        return (double)(maxSum/k);
    }
};